src/Engine/ProtoScript/Runners/LiveRunner.cs

```diff
@@ -254,6 +256,36 @@ private void ApplyChangeSetForceExecute(ChangeSetData changeSet)
             }
         }
 
+        private void ReActivateGraphNodesInCycle(List<AssociativeNode> nodeList)
+        {
+            if (nodeList == null || !nodeList.Any()) return;
+
+            var assocGraph = core.DSExecutable.instrStreamList[0].dependencyGraph;
+            var graphNodes = assocGraph.GetGraphNodesAtScope(Constants.kInvalidIndex, Constants.kInvalidIndex);
+
+            foreach (var node in nodeList)
```

Comment:
will nodeList usually be quite small?

Reply:
These are only those nodes that are modified in a single delta execution run. I guess it could be a long list if the user is in manual mode and has made a large number of edits to their graph before hitting `Run`.

Reply:
It might be worth profiling this change or monitoring the perf benchmark tests after this goes in to see if there is a significant slowdown as a result of this additional check on each delta execution.

Reply:
👍